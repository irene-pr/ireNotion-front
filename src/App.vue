<template>
  <router-view />
</template>

<style lang="scss">
* {
  box-sizing: border-box;
}
body {
  margin: 0;
}
</style>
