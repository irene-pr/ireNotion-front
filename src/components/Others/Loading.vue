<template>
  <div class="loading-page" :class="'loading-page--' + themeSurfaces">
    <div class="circle-animation">
      <div></div>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapState } from "vuex";

export default defineComponent({
  name: "Loading",
  computed: {
    ...mapState(["themeSurfaces"]),
  },
});
</script>

<style lang="scss">
@import "@/styles/_variables";

.loading-page {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 21;
  display: flex;
  justify-content: center;
  align-items: center;

  &--day-mode {
    background-color: $is-loading-background-light;
  }
  &--night-mode {
    background-color: $is-loading-background-dark;
  }
}

.circle-animation {
  display: inline-block;
  transform: translateZ(1px);
}
.circle-animation > div {
  display: inline-block;
  width: 64px;
  height: 64px;
  margin: 8px;
  border-radius: 50%;
  background: $theme-pink;
  animation: circle-animation 2.4s cubic-bezier(0, 0.2, 0.8, 1) infinite;
}
@keyframes circle-animation {
  0%,
  100% {
    animation-timing-function: cubic-bezier(0.5, 0, 1, 0.5);
  }
  0% {
    transform: rotateY(0deg);
  }
  50% {
    transform: rotateY(1800deg);
    animation-timing-function: cubic-bezier(0, 0.5, 0.5, 1);
  }
  100% {
    transform: rotateY(3600deg);
  }
}
</style>
