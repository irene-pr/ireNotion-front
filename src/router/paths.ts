const paths = {
  home: "/",
  login: "/login",
  register: "/register",
  userBoard: "/userBoard",
};

export default paths;
