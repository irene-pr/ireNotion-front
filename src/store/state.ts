const state = {
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
};

export default state;
