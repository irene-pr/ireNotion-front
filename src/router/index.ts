import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";
import paths from "@/router/paths";
import Home from "../views/Home.vue";
import Login from "../views/Login.vue";
import Register from "../views/Register.vue";
import BoardPage from "../views/BoardPage.vue";

const routes: Array<RouteRecordRaw> = [
  {
    path: paths.home,
    name: "Home",
    component: Home,
  },
  {
    path: paths.login,
    name: "Login",
    component: Login,
  },
  {
    path: paths.register,
    name: "Register",
    component: Register,
  },
  {
    path: paths.userBoard,
    name: "BoardPage",
    component: BoardPage,
  },
];

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
});

export default router;
