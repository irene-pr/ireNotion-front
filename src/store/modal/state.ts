import { IModalState } from "@/types/store";

const state = (): IModalState => ({
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
});

export default state;
