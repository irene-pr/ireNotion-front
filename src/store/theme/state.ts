import { IThemeState } from "@/types/store";

const state = (): IThemeState => ({
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
});

export default state;
