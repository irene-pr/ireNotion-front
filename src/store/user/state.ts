import { IUserState } from "@/types/store";

const state = (): IUserState => ({
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  isLoading: false,
});

export default state;
