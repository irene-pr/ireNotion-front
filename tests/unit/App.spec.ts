import { mount } from "@vue/test-utils";
import { createStore } from "vuex";
import App from "@/App.vue";
import Loading from "@/components/Others/Loading.vue";
import state from "@/store/state";
import router from "@/router";

describe("Given an App component", () => {
  const store = createStore({
    state() {
      return state;
    },
  });
  describe("When it renders", () => {
    test("Then it renders", () => {
      mount(App, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });
    });
    test("Then the it matches the snapshot", () => {
      const wrapper = mount(App, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
  describe("When with the initial state", () => {
    test("Then it doesn't render the Loading component", () => {
      const wrapper = mount(App, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      expect(wrapper.findComponent(Loading).exists()).toBeFalsy();
    });
  });
  describe("When with the state isLoading set as true", () => {
    test("Then it renders the Loading component", () => {
      state.isLoading = true;

      const wrapper = mount(App, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      expect(wrapper.findComponent(Loading).exists()).toBeTruthy();
    });
  });
});
