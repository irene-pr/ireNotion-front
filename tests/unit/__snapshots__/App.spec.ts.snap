// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App component When it renders Then the it matches the snapshot 1`] = `
<div
  data-v-app=""
>
  
  <!---->
  <!--v-if-->
  
</div>
`;
